docs/pages/project/contributing/contributing-cli.md

```diff
@@ -128,6 +128,25 @@ When designing for the command line interface, ask and consider the following qu
 
 You will need to provide a short and long description of the command for the help pages and also for the Meshery Documentation.
 
+## Refactoring Mesheryctl Code
+
+The mesheryctl code needs to be refactored for the ease of management, to improve the readbility of the code, for making it easier for using the utility packages. 
+
+For now, the meshconfig has been refactored to have getters and setters which can further be used to manipulate any config. For example, reading or changing the context version has now been a lot easier with the help of specific getters and setters, a central function has also been added to write the config.yaml file.
+
+The workflow for changing the contexts would be:
+- Change the context struct
+- Update the config struct with the new context
+- Write the new config to the config.yaml file
+
+Here are some examples of how the getters and setters of some defined structs are being used:
+
+Let us take an example of the `struct Context`, having the parameter `Token`, the code has been refactored to use `func(ctx *Context) GetToken()` for returning the token of the current context, and `func(ctx *Context) SetToken()` has been used for setting the token of the current context.
+
+Similarly, for the `struct Mesheryctlconfig`, having the `Tokens` parameter has a function named `func (mc *MesheryCtlConfig) GetTokens() *[]Token` which is being used to return the tokens present in the config file.
```

Comment:
Rewrite language - "having the `Tokens`" doesn't flow quite right.

Reply:
```suggestion
Similarly, for the `struct Mesheryctlconfig`, the `Tokens` parameter has a function named `func (mc *MesheryCtlConfig) GetTokens() *[]Token` which is being used to return the tokens present in the config file.
```